tigramite/toymodels/context_model.py

```diff
@@ -0,0 +1,221 @@
+            Keys of the dictionary are from {0, ..., N-1} where N is the number of nodes.
+            Options for the values are "system", "time_context", "space_context". The temporal context variables are
+            assumed exogenous to the system variables. They cannot interact with the spatial context variables due
+            to the assumption that they are constant across datasets.
+        noises : list of callables or array, optional (default: None)
+            Random distribution function that is called with noises[j](T). If an array,
+            it must be of shape ((transient_fraction + 1)*T, N).
+        seed : int, optional (default: None)
+            Random seed.
+
+        Attributes
+        -------
+        links_tc : dict
+            Dictionary of format: {0:[((i, -tau), coeff, func),...], 1:[...],
+            ...} for all temporal context variables where i must be in [0..N-1] and tau >= 0 with
+            number of variables N. coeff must be a float and func a python
+            callable of one argument. The temporal context variables are assumed exogenous
+            to the system variables. They cannot interact with the spatial context variables due to the assumption
+            that they are constant across datasets.
+        links_sc : dict
+            Dictionary of format: {0:[((i, -tau), coeff, func),...], 1:[...],
+            ...} for all spatial context variables where i must be in [0..N-1] and tau >= 0 with
+            number of variables N. coeff must be a float and func a python
+            callable of one argument. The spatial context variables are assumed exogenous
+            to the system variables. They cannot interact with the temporal context variables due to the assumption
+            that they are time-independent, i.e. constant across time.
+        links_sys : dict
+            Dictionary of format: {0:[((i, -tau), coeff, func),...], 1:[...],
+            ...} for all system variables where i must be in [0..N-1] and tau >= 0 with
+            number of variables N. coeff must be a float and func a python
+            callable of one argument.
+        noises : list of callables or array, optional (default: None)
+            Random distribution function that is called with noises[j](T). If an array,
+            it must be of shape ((transient_fraction + 1)*T, N).
+        seed : int, optional (default: None)
+            Random seed.
+
+        """
+
+    def __init__(self, links={}, node_classification={}, noises=None, seed=None):
+        self.links_tc = group_links(links, node_classification, "time_context")
+        self.links_sc = group_links(links, node_classification, "space_context")
+        self.links_sys = group_links(links, node_classification, "system")
+
+        self.N = len(self.links_sys.keys())
+        self.noises = noises
+        self.seed = seed
+
+
+
+    def constant_over_space(self, data_tc, M):
```

Comment:
prefix "_" for all internal functions. If it is public, it should have a complete docstring...